src/queryResult/queryResultWebViewController.ts

```diff
@@ -309,4 +330,31 @@ export class QueryResultWebviewController extends ReactWebviewViewController<
         const messageText = messages.join("\n");
         await this._vscodeWrapper.clipboardWriteText(messageText);
     }
+
+    public getNumExecutionPlanResultSets(
+        resultSetSummaries: qr.QueryResultWebviewState["resultSetSummaries"],
+        actualPlanEnabled: boolean,
+    ): number {
+        const summariesLength = Object.keys(resultSetSummaries).length;
+        if (!actualPlanEnabled) {
+            return summariesLength;
+        }
+        // count the amount of xml showplans in the result summaries
+        let total = 0;
+        Object.values(resultSetSummaries).forEach((batch) => {
+            Object.values(batch).forEach((result) => {
+                // Check if any column in columnInfo has the specific column name
+                if (
+                    result.columnInfo.some(
+                        (column) =>
+                            column.columnName ===
+                            "Microsoft SQL Server 2005 XML Showplan",
```

Comment:
since this string is very specific and is used twice, let's extract it to a shared const.